LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Model/ExtractedProfile.swift

```diff
@@ -166,10 +166,51 @@ struct ExtractedProfile: Codable, Sendable {
             identifier: self.identifier
         )
     }
+
+    /*
+     Matching records are:
+     1/ Completely identical records (same name, addresses, ages, etc)
+     2/ Records that overlap completely (record A has all the data of record B, but might have
+        extra information as well (e.g. an extra address, a middle name where record B doesn't)
+        I.e. B is a subset of A, or vice versa
+     However, we ignore some of the properties
+     So, basically age == age, we ignore phone numbers and email, and then everything else one should be a subset of the other
+     */
+    func doesMatchExtractedProfile(_ extractedProfile: ExtractedProfile) -> Bool {
+        if age != extractedProfile.age {
+            return false
+        }
+
+        if name != extractedProfile.name {
+            return false
+        }
+
+        if !(alternativeNames ?? []).isASubSetOrSuperSetOf(extractedProfile.alternativeNames ?? []) {
+            return false
+        }
+
+        if !(addresses ?? []).isASubSetOrSuperSetOf(extractedProfile.addresses ?? []) {
+            return false
+        }
+
+        if !(relatives ?? []).isASubSetOrSuperSetOf(extractedProfile.relatives ?? []) {
+            return false
+        }
```

Comment:
Prob just a preference, but `guard` statements seem to fit this pattern better IMO (where we return `true` at the end if all conditions pass). Again, maybe just subjective preference.

Reply:
I thought about this and I think I prefer not using guards because to me guard suggests something went wrong or some prerequisite weren't satisfied. I think that is a biased view on my part though and probably not shared universally

Reply:
Fair, one subjective view -vs- another 😅! Fine with me to stick with what you have.